pkg/virt-handler/vm.go

```diff
@@ -2804,18 +2803,13 @@ func isACPIEnabled(vmi *v1.VirtualMachineInstance, domain *api.Domain) bool {
 
 func (d *VirtualMachineController) isHostModelMigratable(vmi *v1.VirtualMachineInstance) error {
 	if cpu := vmi.Spec.Domain.CPU; cpu != nil && cpu.Model == v1.CPUModeHostModel {
-		node, err := d.clientset.CoreV1().Nodes().Get(context.Background(), vmi.Status.NodeName, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		if !nodeHasHostModelLabel(node) {
-			err = fmt.Errorf("the node \"%s\" has no (%s/...) label to allow migration with host-model", node.Name, v1.HostModelCPULabel)
+		// Host CPU model was successfully loaded by NodeLabeller if capabilities is set
+		if d.capabilities == nil {
+			err := fmt.Errorf("the node \"%s\" does not allow migration with host-model", vmi.Status.NodeName)
```

Comment:
@iholder-redhat Please note that currently the NodeLabeller sets the hostCpuModel label unconditionally (https://github.com/kubevirt/kubevirt/blob/b99bc93848db5020fb4983398acecb6888417196/pkg/virt-handler/node-labeller/node_labeller.go#L269).
And only the constant prefix of the label key is verified here: https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/vm.go#L2835
So all we need is the understanding if the node-labeller could detect the host model. As long as no error occurred during the hostCpuModel  load, the values are not important or tested. And in case of an error in loading the values, virt-handler will panic.

Reply:
Great, so your PR won't only make the code better in terms of performance, but also in terms of robustness :+1: